jobs/vip/templates/config/haproxy.conf

```diff
@@ -0,0 +1,62 @@
+<% require "digest"%>
+
+global
+  log 127.0.0.1 syslog <%= p('ha_proxy.log_level') %>
+  daemon
+  user vcap
+  group vcap
+  maxconn 64000
+  spread-checks 4
+  stats socket /var/vcap/sys/run/vip/haproxy.sock mode 600 level admin
+  stats timeout 2m
+  external-check
+
+defaults
+  log global
+  maxconn 64000
+  option contstats
+  timeout connect <%= p("ha_proxy.connect_timeout").to_i * 1000 %>ms
+  timeout client  <%= p("ha_proxy.client_timeout").to_i  * 1000 %>ms
+  timeout server  <%= p("ha_proxy.server_timeout").to_i  * 1000 %>ms
+  timeout queue   <%= p("ha_proxy.queue_timeout").to_i   * 1000 %>ms
+
+
+<% if p("ha_proxy.stats_enable") %>
+listen stats
+  bind <%= p("ha_proxy.stats_bind") %>
+  acl private src <%= p("ha_proxy.trusted_stats_cidrs") %>
+  http-request deny unless private
+  mode http
+  stats enable
+  stats hide-version
+  stats realm Haproxy\ Statistics
+  stats uri /<%= p("ha_proxy.stats_uri") %>
+  stats auth <%= p("ha_proxy.stats_user") %>:<%= p("ha_proxy.stats_password") %>
+<% end %>
+
+frontend postgres
+  mode tcp
+  bind :<%= p('port') %>
+  default_backend postgres
+
+backend postgres
+  mode tcp
+  option external-check
+  external-check command "/var/vcap/jobs/postgres/bin/healthy"
+  external-check path    "/var/vcap/packages/postgres/bin:/bin:/usr/bin:/sbin:/usr/sbin"
+<% link("db").instances.each_with_index do |peer, i| %>
+  server node<%= i %> <%= peer.address %>:<%= p('backend_port') %> check inter 1000<% end %>
+
+
+frontend postgres-read-replica
+mode tcp
+bind :<%= p('postgres.replication.readonly_port') || p('port') + 2000 %>
```

Comment:
Interesting, I never added it to the deployment `.yml` and it had worked/deployed fine. You're right about the specfile. Willfix.

Reply:
You may have to use an `if_p('postgres.replication.readonly_port')` or maybe use `p("postgres.replication.readonly_port", default_value)` instead of the `||` or define it in the spec file.

As per the docs for `p("some.property")` - If `some.property` is not supplied and does not have a default in the spec file, error will be raised to the user specifying that property value is missing

Reply:
Roger, thanks for pointing that out. I'll have all these tickets fixed soon.

Reply:
Don't use `if_p`; the problem is that the spec doesn't have a definition for the property, which is a larger problem.  You can also set the default value in the second argument of `p()`, so you actually want this:

```
bind :<%= p('postgres.replication.readonly_port', p('port') + 2000) %>
```

(I think; try it and see what happens)